### DISCUSSION TITLE
NPX package

### DISCUSSION THREAD

**Discussion by nimit2801** [discordjs/discord.js#6158] [Ideas]
Can we create an npx package for discord.js to bootstrap the process of setting up the package?

```cmd
npx create-discord-bot pokemon-bot
```
Thoughts?

**Comment by nimit2801** [comment]
https://peterthehan.github.io/create-discord-bot/
I got this article online he already has these things published, but something from discord.js would enforce best practices, and we can add this in the discord guide too.

**Comment by advaith1** [comment]
I don't really see the point of this; if you want to quickly get started, just copy-paste the example from the discord.js homepage

**Reply by nimit2801** [reply]
For me I make a lot of community discord bots and setting up everything from scratch is a bit repetitive. So I thought of adding something within the DJs guide so that people can directly take npx packages from there and learn more about bot dev and best practices.

**Comment by splatterxl** [comment]
You could make a GitHub template repository for it, and just select "Use template" when creating a new bot

**Reply by nimit2801** [reply]
Yes, I can but we can use this same template and maybe feature it on the guide or DJs documentation, so that the bot dev can get a good predefined template and just work on the logic of the bot.

**Reply by splatterxl** [reply]
I mean, it's really not necessary from that perspective as you can just `npm init`, install dependencies and start typing in `index.js` (and copy-paste from the docs if needed)! Discord.js isn't really that much of a complicated library to set up.

**Comment by twlite** [comment]
There are some good discord bot template generators such as **[https://github.com/stuyy/slappey](https://github.com/stuyy/slappey)**
which you can run using `npx slappey new <name of project>`
